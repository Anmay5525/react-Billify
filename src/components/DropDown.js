import React from "react";
import PropTypes from "prop-types";

export default function DropDown({ list, handleSelect }) {
  return (
    <select
      style={{
        width: "100%",
        height: "26px",
      }}
      onChange={handleSelect}
    >
      <option className="option">Select Item</option>
      {list.map((item) => {
        return (
          <option
            className="option"
            key={item.name + window.performance.now()}
            value={JSON.stringify(item)}
          >
            {item.name}
          </option>
        );
      })}
    </select>
  );
}

DropDown.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSelect: PropTypes.func.isRequired,
};
