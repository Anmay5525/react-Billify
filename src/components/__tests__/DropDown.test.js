import { mount } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import DropDown from '../DropDown';

describe('Testing DropDown component if it', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DropDown list={[]} handleSelect={() => {}}></DropDown>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const container = mount(<DropDown list={[{name: 'David'}, {name: 'John'}]} handleSelect={() => {}}></DropDown>);
    expect(container.debug()).toMatchSnapshot();
  });

})