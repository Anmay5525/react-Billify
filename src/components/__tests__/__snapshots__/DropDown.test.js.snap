// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DropDown component if it matches snapshot 1`] = `
"<DropDown list={{...}} handleSelect={[Function: handleSelect]}>
  <select style={{...}} onChange={[Function: handleSelect]}>
    <option className=\\"option\\">
      Select Item
    </option>
    <option className=\\"option\\" value=\\"{\\"name\\":\\"David\\"}\\">
      David
    </option>
    <option className=\\"option\\" value=\\"{\\"name\\":\\"John\\"}\\">
      John
    </option>
  </select>
</DropDown>"
`;
